### DISCUSSION TITLE
dynamic loading images

### DISCUSSION THREAD

**Discussion by gregorvoinov** [nuxt/nuxt#16593] [Questions] [answered]
hi,

I have started with nuxt3 and am having some problems loading dynamic images. 

I have found this answer many times in my research, but it does not work for me

```
<template>
  <img :src="img_src" alt="">
</template>

computed: {
  img_src() {
    return new URL(`../../../assets/images/${this.file}`, import.meta.url).href`
  }
}
```
I also tried to use "~/assets/assets/images/${this.file}" but also doesn't worked

I get only this error

`Not allowed to load local resource: file:///Users/gregor/Development/nuxt3/assets/images/img4.jpg`

is there an other way to load dynamic images or what's wrong with this approach?

**UPDATE**
Found also this solution

https://github.com/vitejs/vite/issues/1265

but didn't work again for me :/

```
    const getImg = async file => {
        return await import(/* @vite-ignore */ `../../../assets/images/${file}`).default
        // const module = await import(/* @vite-ignore */ `../../../assets/images/${file}`).default
        // return module.replace(/^\/@fs/, '')
    }

    watchEffect(async () => (
        img.value = await getImg(props.file)
    ))
```

with replace 'fs' I get this

```
  const module = await import(/* @vite-ignore */ `../../../assets/images/${file}`)
  return module.replace(/^\/@fs/, '')
```
<img width="848" alt="image" src="https://user-images.githubusercontent.com/9499590/193504545-c1073560-3d41-4364-8da7-26893cca99fe.png">

or with this line only

```
  return await import(/* @vite-ignore */ `../../../assets/images/${file}`).default
```

I get this error

<img width="565" alt="image" src="https://user-images.githubusercontent.com/9499590/193504815-19b18ce0-a966-43e4-bffa-c768d126dab5.png">

When I click on the link of the image in the error message, the image opens

best gregor

**Comment by gregorvoinov** [comment]
if I put the file in the public folder I can easily us it with

```
<img :src="`/images/${props.file}`" />
```

But no idea how to use it from the assets folder...

**Reply by echo9z** [reply]
can use vue3-lazyload：https://www.npmjs.com/package/vue3-lazyload

**Comment by motea927** [comment, marked as answer]
Hi, you can reference some document

1. `Assets` folder needs a process by `build tool`. (Ref: https://v3.nuxtjs.org/guide/directory-structure/assets)
2. `new URL` Does not work with SSR. (Ref: https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url)
3. Dynamic import limitations. (Ref: https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations)

So, If you want use dynamic assets, you need let build tool know what's file may import in your project (like start with `./` or `../` or Imports must end with a file extension)

I recommend using the public folder in development, and use another online CDN url in production,
But there is an example to achieve your goal (Ref: https://stackblitz.com/edit/github-qiedxz?file=app.vue)

**Comment by gregorvoinov** [comment]
thank you for the links got it running with

```
watchEffect(async () => {
    importImage.value = await import(
      `../assets/images/${props.file}.jpg`
    ).then((lib) => lib.default || lib)
```

but for simplicity I will use the public folder for images, if there is no downside.

**Reply by TheAlexLichter** [reply]
Downside would be no cache busting due to missing asset hashes.

**Reply by lacorde** [reply]
Anyone could explain the use of watchEffect in that case ? Thanks for the example @motea927.
